assembl/scripts/ini_files.py

```diff
@@ -144,72 +166,392 @@ def main():
-        if not exists(join(vroot_lib, 'virtodbcu.so'))\
-                and exists(join(vroot_lib, 'odbc', 'virtodbcu.so')):
-            vroot_lib = join(vroot_lib, 'odbc')
-        vname = 'virtuoso'
-        if not exists(join(vroot, 'share', vname)):
-            names = listdir(join(vroot, 'share'))
-            names = [n for n in names
-                     if exists(join(vroot, 'share', n, 'vad'))]
-            assert len(names) == 1, "Cannot identify the vad directory"
-            vname = names[0]
-        assert exists(join(vroot_var, 'lib', vname, 'vsp')),\
-            "Cannot identify the VSP directory"
-        vars.update({
-            'VIRTUOSO_SERVER_PORT': config.getint('virtuoso', 'http_port'),
-            'VIRTUOSO_HOSTNAME': config.get(SECTION, 'public_hostname'),
-            'VIRTUOSO_PORT': config.getint('virtuoso', 'port'),
-            'VIRTUOSO_ROOT': vroot,
-            'VIRTUOSO_ROOT_VAR': vroot_var,
-            'VIRTUOSO_ROOT_LIB': vroot_lib,
-            'VIRTUOSO_SUBDIR_NAME': vname,
-        })
-    else:
-        # dummy values
-        vars.update({
-            'VIRTUOSO_SERVER_PORT': 8890,
-            'VIRTUOSO_HOSTNAME': 'localhost',
-            'VIRTUOSO_PORT': 1111,
-            'VIRTUOSO_ROOT': '/usr',
-            'VIRTUOSO_ROOT_VAR': '/usr/var',
-            'VIRTUOSO_ROOT_LIB': '/usr/var/lib',
-            'VIRTUOSO_SUBDIR_NAME': '/usr/share/virtuoso/vad',
-        })
-
-    print vars
-    for fname in ('var/db/virtuoso.ini', 'odbc.ini', 'supervisord.conf',):
+
+    for fname in ('supervisord.conf',):
         print fname
-        tmpl = open(fname+'.tmpl').read()
-        inifile = open(fname, 'w')
-        inifile.write(tmpl % vars)
-        inifile.close()
+        with open(fname + '.tmpl') as tmpl, open(fname, 'w') as inifile:
+            inifile.write(tmpl.read() % vars)
+
+
+def combine_ini(config, overlay, adding=True):
+    """Overlay values of the second ini file on the first.
+
+    Returns a ConfigParser object
+    If adding is false, only existing values will be replaced.
+    """
+    config = asParser(config)
+    overlay = asParser(overlay)
+    for section in overlay.sections():
+        # Avoid including DEFAULTSECT
+        for key in overlay._sections[section]:
+            value = overlay.get(section, key)
+            if adding or config.has_option(section, key):
+                ensureSection(config, section)
+                config.set(section, key, value)
+    if overlay._defaults:
+        config._defaults.update(overlay._defaults)
+    return config
+
+
+def migrate_beaker_config(random_ini, overlay):
+    """Migrate old-style session... to beaker.session..."""
+    for section in random_ini.sections():
+        # Avoid including DEFAULTSECT
+        if not overlay.has_section(section):
+            continue
+        o_section = overlay._sections[section]
+        for key in random_ini._sections[section]:
+            if key.startswith("beaker."):
```

Comment:
This is hackish, yes. But it's about removing legacy complexity in a migration, not the main code.